PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/TelemetryClientProvider.java

```diff
@@ -0,0 +1,333 @@
+
+import java.io.FileNotFoundException;
+
+import com.intellij.psi.PsiVariable;
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * This class reads the instrumentation key from the applicationInsights.json file
+ * and returns a TelemetryClient object with the instrumentation key set.
+ * This object is used to send telemetry data to Application Insights.
+ */
+public class TelemetryClientProvider extends LocalInspectionTool {
+
+    /**
+     * This method is called by the IntelliJ platform to build a visitor for the inspection.
+     *
+     * @param holder     The ProblemsHolder object that holds the problems found in the code.
+     * @param isOnTheFly A boolean that indicates if the inspection is running on the fly.
+     * @return A PsiElementVisitor object that visits the method calls in the code.
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        // Reset the methodCounts map for each new visitor
+        TelemetryClientProviderVisitor.methodCounts.clear();
+        TelemetryClientProviderVisitor visitor = new TelemetryClientProviderVisitor(holder, isOnTheFly);
+
+        // Set the visitor instance in TelemetryToggleAction
+        TelemetryToggleAction.setTelemetryService(visitor);
+
+        return visitor;
+    }
+
+    /**
+     * This class is a visitor that visits the method calls in the code and tracks the method calls.
+     */
+    static class TelemetryClientProviderVisitor extends JavaElementVisitor {
+
+        private final ProblemsHolder holder;
+        private static boolean running = false; // Flag to indicate if the telemetry service is running
+        private static ScheduledExecutorService executorService; // Executor service to schedule telemetry data sending
+
+        // Create a TelemetryClient object
+        // not final because the test involves Injecting the mock telemetry client to telemetryClient
+        // Package-private to allow access from tests in the same package
+        static TelemetryClient telemetryClient = getTelemetryClient();
+        // Create a map to store the method counts
+        static Map<String, Map<String, Integer>> methodCounts = new HashMap<>();
+
+        // Create a Project object
+        private static Project project;
+
+        // Create a list of prefixes for Azure service method calls
+        // source: https://azure.github.io/azure-sdk/java_introduction.html#service-methods
```

Comment:
/** 
 * Refer to <a href="https://azure.github.io/azure-sdk/java_introduction.html#service-methods">Your data</a> 
 */
 
 Correct way to add hyperlink in comment.